src/crimson/osd/recovery_backend.h

```diff
@@ -215,9 +215,9 @@ class RecoveryBackend {
     void set_pushed(pg_shard_t shard) {
       auto it = pushes.find(shard);
       if (it != pushes.end()) {
-	auto &push_promise = it->second;
-	push_promise.set_value();
-	pushes.erase(it);
+        auto &push_promise = it->second;
```

Comment:
Going back to my first comment, as both `maybe_push_shards` and `push_delete` are updated and are the **only** users of set_pushed. Can we assert `pushes.contains(shard)` here? Is there any other reason to keep the `if` instead?

Reply:
done

Reply:
Thanks! If we assert it there is no need to also having the `if` right? we can use `pushes.at`

Reply:
sure